From 760c25c43d1175a429482dc9e59b59f7c3a998de Mon Sep 17 00:00:00 2001
From: gord chung <gord@live.ca>
Date: Fri, 8 Dec 2017 19:33:02 +0000
Subject: [PATCH 75/91] cleanup data-pipelines admin guide

- add a note mentioning transformers are not durable and you
might be able to use gnocchi
- slight change to cherry-pick, as we can't enable/disable pipelines
in pike

Change-Id: I3f50d5b5a1e22a114ef2c8e17c1e659f1c0eb761
(cherry picked from commit 935b53c9df6d2265cba7e58a19d9b82d67acb220)
---
 doc/source/admin/telemetry-data-pipelines.rst | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/doc/source/admin/telemetry-data-pipelines.rst b/doc/source/admin/telemetry-data-pipelines.rst
index c882ca4..216c819 100644
--- a/doc/source/admin/telemetry-data-pipelines.rst
+++ b/doc/source/admin/telemetry-data-pipelines.rst
@@ -34,11 +34,7 @@ The pipeline configuration is, by default stored in separate configuration
 files called ``pipeline.yaml`` and ``event_pipeline.yaml`` next to
 the ``ceilometer.conf`` file. The meter pipeline and event pipeline
 configuration files can be set by the ``pipeline_cfg_file`` and
-``event_pipeline_cfg_file`` options listed in the `Description of
-configuration options for api table
-<https://docs.openstack.org/ocata/config-reference/telemetry/telemetry-config-options.html>`__
-section in the OpenStack Configuration Reference respectively. Multiple
-pipelines can be defined in one pipeline configuration file.
+``event_pipeline_cfg_file`` options listed in :ref:`configuring`.
 
 The meter pipeline definition looks like:
 
@@ -146,6 +142,12 @@ The event filter uses the same filtering logic as the meter pipeline.
 Transformers
 ------------
 
+.. note::
+
+   Transformers maintain data in memory and therefore do not guarantee
+   durability in certain scenarios. A more durable and efficient solution
+   may be achieved post-storage using solutions like Gnocchi.
+
 The definition of transformers can contain the following fields:
 
 name
@@ -380,13 +382,6 @@ be configured for each data point within the Telemetry service, allowing
 the same technical meter or event to be published multiple times to
 multiple destinations, each potentially using a different transport.
 
-Publishers are specified in the ``publishers`` section for each
-pipeline that is defined in the `pipeline.yaml
-<https://git.openstack.org/cgit/openstack/ceilometer/plain/ceilometer/pipeline/data/pipeline.yaml>`__
-and the `event_pipeline.yaml
-<https://git.openstack.org/cgit/openstack/ceilometer/plain/ceilometer/pipeline/data/event_pipeline.yaml>`__
-files.
-
 The following publisher types are supported:
 
 gnocchi (default)
-- 
2.7.4

